clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapInstanceConfig.java

```diff
@@ -40,73 +40,88 @@ public CleverTapInstanceConfig[] newArray(int size) {
     };
 
     private String accountId;
-
     private String accountRegion;
-
     private String accountToken;
-
     private String proxyDomain;
-
     private String spikyProxyDomain;
-
     private String customHandshakeDomain;
-
-    @NonNull
-    private ArrayList<String> allowedPushTypes = getAll();
-
+    @NonNull private ArrayList<String> allowedPushTypes = getAll();
     private boolean analyticsOnly;
-
     private boolean backgroundSync;
-
     private boolean beta;
-
     private boolean createdPostAppLaunch;
-
     private int debugLevel;
-
     private boolean disableAppLaunchedEvent;
-
     private boolean enableCustomCleverTapId;
-
     private String fcmSenderId;
-
     private boolean isDefaultInstance;
-
     private Logger logger;
-
     private String packageName;
-
     private boolean personalization;
-
     private String[] identityKeys = Constants.NULL_STRING_ARRAY;
-
     private boolean sslPinning;
-
     private boolean useGoogleAdId;
     private int encryptionLevel;
 
-
     @SuppressWarnings("unused")
-    public static CleverTapInstanceConfig createInstance(Context context, @NonNull String accountId,
-            @NonNull String accountToken) {
+    public static CleverTapInstanceConfig createInstance(
+            Context context,
+            @NonNull String accountId,
+            @NonNull String accountToken
+    ) {
+        return CleverTapInstanceConfig.createInstance(context, accountId, accountToken, null);
+    }
 
+    @SuppressWarnings({"unused"})
+    public static CleverTapInstanceConfig createInstance(
+            @NonNull Context context,
+            @NonNull String accountId,
+            @NonNull String accountToken,
+            @Nullable String accountRegion
+    ) {
         //noinspection ConstantConditions
         if (accountId == null || accountToken == null) {
             Logger.i("CleverTap accountId and accountToken cannot be null");
             return null;
         }
-        return new CleverTapInstanceConfig(context, accountId, accountToken, null, false);
+        ManifestInfo manifestInfo = ManifestInfo.getInstance(context);
+        return CleverTapInstanceConfig.createInstanceWithManifest(manifestInfo, accountId, accountToken, accountRegion, false);
     }
 
-    @SuppressWarnings({"unused"})
-    public static CleverTapInstanceConfig createInstance(Context context, @NonNull String accountId,
-            @NonNull String accountToken, String accountRegion) {
-        //noinspection ConstantConditions
-        if (accountId == null || accountToken == null) {
-            Logger.i("CleverTap accountId and accountToken cannot be null");
+
+    // convenience to construct the internal only default config
+    @SuppressWarnings({"unused", "WeakerAccess"})
+    protected static CleverTapInstanceConfig createDefaultInstance(
```

Comment:
The access modifiers to this method and `createInstanceWithManifest` are a bit weird. This is protected, but the other is public and can achieve the same thing.

Reply:
fixed visibility of createInstanceWithManifest